repository/repository.go

```diff
@@ -203,3 +203,24 @@ func (r *Repository) Delete(ctx context.Context, id string) error {
 	}
 	return nil
 }
+
+func (r *Repository) Checkout(ctx context.Context, id string) (string, error) {
```

Comment:
this makes me wanna have a separate UserRepository type...

in my head the ideal Repository is centered around the remote, whereas this method is way more centered around the UserRepository.